test/cql-pytest/test_allow_filtering.py

```diff
@@ -291,3 +291,23 @@ def test_contains_frozen_collection_ck(cql, test_keyspace):
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS 0 ALLOW FILTERING")))
         assert 1 == len(list(cql.execute(
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS KEY 0 ALLOW FILTERING")))
+
+# table4 contains example table from issue #8991
+@pytest.fixture(scope="module")
+def table4(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (k int, c1 int, c2 int, PRIMARY KEY (k,c1,c2))")
+    cql.execute(f"CREATE INDEX ON {table} (c2)")
+    for c1 in [0, 1, 2]:
+        for c2 in [0, 1, 2]:
+            cql.execute(f"INSERT INTO {table} (k, c1, c2) VALUES (0, {c1}, {c2})")
+    everything = list(cql.execute(f"SELECT * FROM {table}"))
+    wait_for_index(cql, table, 'c2', everything)
+    yield (table, everything)
+    cql.execute(f"DROP TABLE {table}")
```

Comment:
I wonder if you should move the test you put in test_secondary_indexing.py into this file, to reuse the same table4 instead of creating it twice.

Reply:
Should I do it?
I didn't want to add tests that don't check if allow filtering is required to `test_allow_filtering.py`
`test_secondary_indexing.py` seems to better match checking correctness of indexed queries

Reply:
No, what you did is fine and your reasoning is correct. Creating yet another table has a cost (about 0.1 seconds) but we can live with it. So I'm resolving this discussion.